assembly-main/src/assembly/bin/che-install-plugin.sh

```diff
@@ -10,34 +10,309 @@
+  ### Set the value of derived environment variables.
+  ### Use values set by user, unless they are broken, then fix them
+  # The base directory of Che
+  if [ -z "${CHE_HOME}" ]; then
+    export CHE_HOME="$(dirname "$(cd $(dirname ${0}) && pwd -P)")"
+  fi
+
+  # Convert Tomcat environment variables to POSIX format.
+  if [[ "${JAVA_HOME}" == *":"* ]]
+  then 
+    JAVA_HOME=$(echo /"${JAVA_HOME}" | sed  's|\\|/|g' | sed 's|:||g')
+  fi
+
+  if [[ "${CHE_HOME}" == *":"* ]]
+  then 
+    CHE_HOME=$(echo /"${CHE_HOME}" | sed  's|\\|/|g' | sed 's|:||g')
+  fi
+
+  # Where 3rd party extensions are staged
+  PLUGIN_DIR="${CHE_HOME}/plugins"
+  SDK_DIR="${CHE_HOME}/sdk"
+
+  PLUGIN_IDE_DIR="${PLUGIN_DIR}/ide"
+  PLUGIN_WSMASTER_DIR="${PLUGIN_DIR}/ws-master"
+  PLUGIN_WSAGENT_DIR="${PLUGIN_DIR}/ws-agent"
+  
+  # The IDE project that will be built with the extension
+  PLUGIN_IDE_WAR_DIR="${SDK_DIR}/assembly-ide-war"
+
+  # The machine web app project that will be built with the extension
+  PLUGIN_MACHINE_WAR_DIR="${SDK_DIR}/assembly-machine-war"
+
+  # Creates the ws-agent.zip artifact for a Che assembly, which packages Tomcat + machine war into single package
+  PLUGIN_MACHINE_SERVER_DIR="${SDK_DIR}/assembly-machine-server"
+
+  # Generates a new Che assembly that contains new IDE, ws-master, and ws-agent.
+  PLUGIN_ASSEMBLY_DIR="${SDK_DIR}/assembly-main"
+
+}
+
+function echo_stage {
+
+    echo
+    echo "#####################################################################"
+    echo "$1"
+    echo "#####################################################################"
+    echo
+
+}
+
+init_global_variables
+parse_command_line "$@"
+set_environment_variables
+
+if [ "${USE_HELP}" == "false" ]; then
+
+
+
+  if [ "${SKIP_MAVEN}" == "false" ]; then
+    echo_stage "CHE SDK: Installing each extension into local maven repository"
+  fi
+
+  # Install every 3rd-party extension into local Maven repository
+  for file in $PLUGIN_DIR/*.jar
+  do
     if [ -f $file ]; then
-        mvn org.apache.maven.plugins:maven-install-plugin:2.5.1:install-file -Dfile=$file
+        cp $file $PLUGIN_IDE_DIR
+        cp $file $PLUGIN_CHE_DIR
+        cp $file $PLUGIN_WORKSPACE_DIR
+      
+        if [ "${SKIP_MAVEN}" == "false" ]; then
+          mvn org.apache.maven.plugins:maven-install-plugin:2.5.1:install-file -Dfile=$file
+        fi
+    fi
+  done
+  
+
+  if [ "${SKIP_DEPENDENCIES}" == "false" ]; then
+
+    echo_stage "CHE SDK: Adding IDE extensions as dependencies"
+
+    # Prepare to re-build Codenvy IDE
+    java -cp "${CHE_HOME}/sdk/che-plugin-sdk-tools.jar" org.eclipse.che.ide.sdk.tools.InstallExtension --extDir="${PLUGIN_IDE_DIR}" --extResourcesDir="${PLUGIN_IDE_WAR_DIR}"
+    #java -cp c:/codenvy/che-plugins/plugin-sdk/che-plugin-sdk-tools/target/che-plugin-sdk-tools-4.0.0-beta-8-SNAPSHOT-jar-with-dependencies.jar org.eclipse.che.ide.sdk.tools.InstallExtension --extDir="${PLUGIN_IDE_DIR}" --extResourcesDir="${PLUGIN_IDE_WAR_DIR}"
```

Comment:
No.  They are a hangover.  Forgot to remove them after my debugging was done.  Will update.